### PR TITLE
accounts/abi/bind Handle structs used only as constructor params

### PR DESCRIPTION
**Pull request by divergencetech**
Fixes issue #23939

The `structs` map is populated by iterating over all methods except the constructor,
which results in a nil-pointer dereference.

I've first reproduced the problem with a new test and then implemented the fix.

### CODE REVIEW

File: accounts/abi/bind/bind_test.go [hunk 1]
```diff
@@ -1934,22 +1978,23 @@ func TestGolangBindings(t *testing.T) {
...
-		bind, err := Bind(types, tt.abi, tt.bytecode, tt.fsigs, "bindtest", LangGo, tt.libs, tt.aliases)
-		if err != nil {
-			t.Fatalf("test %d: failed to generate binding: %v", i, err)
-		}
-		if err = ioutil.WriteFile(filepath.Join(pkg, strings.ToLower(tt.name)+".go"), []byte(bind), 0600); err != nil {
-			t.Fatalf("test %d: failed to write binding: %v", i, err)
-		}
-		// Generate the test file with the injected test code
-		code := fmt.Sprintf(`
+		t.Run(tt.name, func(t *testing.T) {
```
Comment (divergencetech) [inline accounts/abi/bind/bind_test.go:1981]: The diff here suggests more substantial changes than I've actually performed. I've merely run each test as a sub-test for improved reporting as they're easier to read than `test %d`.

**Review by MariusVanDerWijden — APPROVED** [review]
LGTM

**Comment by divergencetech** [comment]
Is it possible to rerun the checks? The failing test looks spurious and unrelated. I've run `make test` on my branch without issue.

**Comment by holiman** [comment]
> Is it possible to rerun the checks? 

Doing that now

**Pull request merged**